app/controllers/admin/cms/transcripts_controller.rb

```diff
@@ -30,12 +30,35 @@ def update
     end
   end
 
+  def speaker_search
+    speakers = Speaker.where("LOWER(name) LIKE ?", "%#{params[:q].downcase}%").map {|s|
+      {id: s.id, label: s.name, value: s.name}
+    }
+    render json: speakers
+  end
+
   private
 
   def set_transcript
     @transcript = Transcript.find_by(uid: params[:id])
   end
 
+  def set_speakers(transcript)
```

Comment:
no bueno, we want to keep controllers logic-free. controllers are for receiving requests, passing params into _other_ classes to do the work, and then redirecting/rendering a response

does it work if you define a method on the model called `def speakers=(speaker_params)`? that should receiving the `transcript_params[:speakers]` hash and you can then do this logic in there (and write a spec for it)